anndata/_io/specs/methods.py

```diff
@@ -562,6 +562,12 @@ def write_sparse_dataset(f, k, elem, _writer, dataset_kwargs=MappingProxyType({}
 )
 def write_dask_sparse(f, k, elem, _writer, dataset_kwargs=MappingProxyType({})):
     sparse_format = elem._meta.format
+
+    def with_int64_indptr(x):
+        x.indptr = x.indptr.astype(np.int64)
+        return x
+
+    elem = elem.map_blocks(with_int64_indptr, dtype=elem.dtype)
```

Comment:
```suggestion
    def with_uint64_indptr(x):
        x.indptr = x.indptr.astype(np.uint64)
        return x

    elem = elem.map_blocks(with_uint64_indptr, dtype=elem.dtype)
```
this should be ok, no? and a nice space saver?  you can construct a sparse matrix and access it with 64 bit integer indices, but that's about it.  luckily, that seems to be enough for this!  it seemed to work locally for me!

Reply:
I know there had been discussion about numpy's "very bad" promotion rules for unsigned integers, but this is just local, no?